scene/gui/label.cpp

```diff
@@ -381,73 +475,95 @@ Rect2 Label::get_line_rect(int p_line) const {
 			}
 		} break;
 	}
-
 	return Rect2(offset, line_size);
 }
 
-void Label::get_layout_data(Vector2 &r_offset, int &r_line_limit, int &r_line_spacing) const {
+int Label::get_layout_data(Vector2 &r_offset, int &r_last_line, int &r_line_spacing) const {
 	// Computes several common parameters involved in laying out and rendering text set to this label.
 	// Only vertical margin is considered in r_offset: use get_line_rect to get the horizontal offset
 	// for a given line of text.
 	Size2 size = get_size();
 	Ref<StyleBox> style = theme_cache.normal_style;
 	int line_spacing = settings.is_valid() ? settings->get_line_spacing() : theme_cache.line_spacing;
-
+	int paragraph_spacing = settings.is_valid() ? settings->get_paragraph_spacing() : theme_cache.paragraph_spacing;
 	float total_h = 0.0;
 	int lines_visible = 0;
 
 	// Get number of lines to fit to the height.
-	for (int64_t i = lines_skipped; i < lines_rid.size(); i++) {
-		total_h += TS->shaped_text_get_size(lines_rid[i]).y + line_spacing;
-		if (total_h > (get_size().height - style->get_minimum_size().height + line_spacing)) {
-			break;
+	int line_index = 0;
+	for (const Paragraph &para : paragraphs) {
+		if (line_index + para.lines_rid.size() <= lines_skipped) {
+			line_index += para.lines_rid.size();
+		} else {
+			int start = (line_index < lines_skipped) ? lines_skipped - line_index : 0;
+			for (int i = start; i < para.lines_rid.size(); i++) {
+				total_h += TS->shaped_text_get_size(para.lines_rid[i]).y + line_spacing;
```

Comment:
```suggestion
			for (const RID &line_rid : para.lines_rid) {
				total_h += TS->shaped_text_get_size(line_rid).y + line_spacing;
```

Reply:
It's not iterating over a full range here.